site/content/blog/2020-06-04-zap-2-9-0-highlights/index.md

```diff
@@ -0,0 +1,174 @@
+var ScriptVars = Java.type('org.zaproxy.zap.extension.script.ScriptVars');
+
+function extractWebSession(sessionWrapper) {
+	// parse the authentication response
+	var json = JSON.parse(sessionWrapper.getHttpMessage().getResponseBody().toString());
+	var token = json.authentication.token;
+	// save the authentication token
+	sessionWrapper.getSession().setValue("token", token);
+	ScriptVars.setGlobalVar("juiceshop.token", token);
+}
+    	
+function clearWebSessionIdentifiers(sessionWrapper) {
+	var headers = sessionWrapper.getHttpMessage().getRequestHeader();
+	headers.setHeader("Authorization", null);
+	ScriptVars.setGlobalVar("juiceshop.token", null);
+}
+    	
+function processMessageToMatchSession(sessionWrapper) {
+	var token = sessionWrapper.getSession().getValue("token");
+	if (token === null) {
+		print('JS mgmt script: no token');
+		return;
+	}
+	var cookie = new HtmlParameter(COOKIE_TYPE, "token", token);
+	// add the saved authentication token as an Authentication header and a cookie
+	var msg = sessionWrapper.getHttpMessage();
+	msg.getRequestHeader().setHeader("Authorization", "Bearer " + token);
+	var cookies = msg.getRequestHeader().getCookieParams();
+	cookies.add(cookie);
+	msg.getRequestHeader().setCookieParams(cookies);
+}
+
+function getRequiredParamsNames() {
+	return []; 
+}
+
+function getOptionalParamsNames() {
+	return [];
+}
+```
+
+The example starts out with a documentation block explaining the assumptions for the Context setup and user being authenticated. The next few lines define the 
+imports/types that will be used by the script. Then we get to the business part of the script.
+
+The `extractWebSession` function extracts the token value and saves it into a Global variable where it can later be retrieved by other scripts/functions.
+
+The next function `clearWebSessionIdentifiers` is setup to remove or invalidate a token from within ZAP.
+
+`processMessageToMatchSession` is responsible for inserting the token where necessary, in the case of Juice Shop both the `Authorization` header value and a 
+cookie value.
+
+The functions `getRequiredParamsNames` and `getOptionalParamsNames` are un-used for Juice Shop session management.
+These functions are called during the script loading to obtain a list of the names of the required (or optional, as applicable) configuration parameters, 
+that will be shown in the Session Properties -> Session Management panel for configuration. They can be used to input dynamic data into the script, 
+from the user interface (e.g. a JSONPath expression for extracting values from a JSON response, the name of a header, a login URL, or name of POST parameters, etc.)
+
+For more information about getting familiar with ZAP and setting up/managing authentication for ZAP scans refer to the 
+[ZAP in Ten](https://www.alldaydevops.com/zap-in-ten) series.
+
+## Conclusion
+
+For further details see the [2.9.0 release notes](/docs/desktop/releases/2.9.0/).
```

Comment:
Not really a conclusion ;) And this blog post actually gives more details about many of the 2.9.0 features than the release notes do!
Could maybe something about ZAP continually improving, this just goes into more details on a few of the changes and for the full set of changes see the release notes?

Reply:
Good point.